ipu-plugin/pkg/ipuplugin/lifecycleservice.go

```diff
@@ -230,7 +254,248 @@ func configureChannel(mode, daemonHostIp, daemonIpuIp string) error {
+	fi
+	`, macAddress) // Insert the MAC address variable into the script.
+
+	commands := fmt.Sprintf(`
+	cat <<'EOF' > /work/scripts/load_custom_pkg.sh
+	%s
+	EOF
+	`, shellScript)
+
+	uuidFilePath := "/work/uuid"
+	uuidFile, err := sftpClient.Create(uuidFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to create remote uuid file: %s", err)
+	}
+	defer uuidFile.Close()
+
+	// Write the new MAC address to the uuid file.
+	_, err = uuidFile.Write([]byte(macAddress + "\n"))
+	if err != nil {
+		return fmt.Errorf("failed to write to uuid file: %s", err)
+	}
+
+	// Ensure that the uuid file is written to the remote filesystem.
+	err = uuidFile.Sync()
+	if err != nil {
+		return fmt.Errorf("failed to sync uuid file: %s", err)
+	}
+
+	// Run a command on the remote server and capture the output.
+	var stdoutBuf bytes.Buffer
+	session.Stdout = &stdoutBuf
+	err = session.Run(commands)
+	if err != nil {
+		return fmt.Errorf("failed to run commands: %s", err)
+	}
+
+	session, err = client.NewSession()
+	if err != nil {
+		return fmt.Errorf("failed to create session: %s", err)
+	}
+	defer session.Close()
+
+	err = session.Run("reboot")
+	if err != nil {
+		return fmt.Errorf("failed to run commands: %s", err)
+	}
+
+	return nil
+}
+
+func countAPFDevices() int {
+	var pfList []netlink.Link
+
+	if err := getFilteredPFs(&pfList); err != nil {
+		return 0
+	}
+
+	return len(pfList)
+}
+
+func checkMAC() (bool, string) {
+	config := &ssh.ClientConfig{
+		User: "root",
+		Auth: []ssh.AuthMethod{
+			ssh.Password(""),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+
+	// Connect to the remote server.
+	client, err := ssh.Dial("tcp", "192.168.0.1:22", config)
+	if err != nil {
+		return false, fmt.Sprintf("failed to dial remote server: %s", err)
+	}
+	defer client.Close()
+
+	// Start a session.
+	session, err := client.NewSession()
+	if err != nil {
+		return false, fmt.Sprintf("failed to create session: %s", err)
+	}
+	defer session.Close()
+
+	commands := "if [ -f /work/uuid ]; then echo 'File exists'; else echo 'File does not exist'; fi"
+
+	// Run a command on the remote server and capture the output.
+	var stdoutBuf bytes.Buffer
+	session.Stdout = &stdoutBuf
+	err = session.Run(commands)
+	if err != nil {
+		return false, fmt.Sprintf("mac not found: %s", err)
+	}
+
+	output := stdoutBuf.String()
+	fmt.Println(output)
+	if output == "File exists\n" {
+		return true, "File exists"
+	} else {
+		return false, "File does not exist"
+	}
+}
+
+func (e *ExecutableHandlerImpl) validate() bool {
+
+	if numAPFs := countAPFDevices(); numAPFs < 19 {
+		fmt.Printf("Not enough APFs %v", numAPFs)
+		return false
+	}
+
+	if macPreFix, mac := checkMAC(); !macPreFix {
+		fmt.Printf("incorrect Mac assigned : %v\n", mac)
+		return false
+	}
+
+	return true
+}
+
 func (s *LifeCycleServiceServer) Init(ctx context.Context, in *pb.InitRequest) (*pb.IpPort, error) {
+	initHandlers()
+
+	if val := executableHandler.validate(); !val {
```

Comment:
this should only be run if we are in IPU Mode, Host mode will be different.
I will add the check, and then a separate PR can be raised for host side validation